### DISCUSSION TITLE
Unable to deploy in Netlify

### DISCUSSION THREAD

**Discussion by torvenkat** [thuliteio/doks#1162] [How-to guides] [answered]
I am using the latest build v1.2.0 of doks. I could locally build everything successfully.  When I try to deploy at Netlify (using my github repository), I get the following error (I set up the build command in Netlify as `npm run build`)

```
11:35:10 AM: .../node_modules/gethyas postinstall: - Architecture  x64
11:35:10 AM: .../node_modules/gethyas postinstall: hugo v0.119.0-b84644c008e0dc2c4b67bd69cccf87a41a03937e+extended linux/amd64 BuildDate=2023-09-24T15:20:17Z VendorInfo=gohugoio
11:35:10 AM: .../node_modules/gethyas postinstall: Success!
11:35:10 AM: .../node_modules/gethyas postinstall: Done
11:35:10 AM: > tam@0.0.0 postinstall /opt/build/repo
11:35:10 AM: > shx rm -rf node_modules/.bin/hugo && shx mkdir node_modules/.bin/hugo && shx cp node_modules/gethyas/node_modules/.bin/hugo/* node_modules/.bin/hugo

sh: 1: shx: not found
11:35:10 AM:  ELIFECYCLE  Command failed.
11:35:10 AM: Error during pnpm install
11:35:10 AM: Failing build: Failed to install dependencies
```
Can you please help?

I also tried using both `hyas build` and `npm run build`.  I also tried using 'npm run build' in '[build]' section as 
```
  command = "npm run build"
  publish = "public"
```
It fails.

**Comment by h-enk** [comment]
Could you please share your repo?

**Reply by torvenkat** [reply]
Thanks for the quick response.  It is https://github.com/torvenkat/eng

**Reply by h-enk** [reply, marked as answer]
I noticed you're missing a bunch of files in your project root.

One is critical — add `.npmrc` containing:

```ini
enable-pre-post-scripts = true
auto-install-peers = true
node-linker = hoisted
prefer-symlinked-executables = false
```

You should also add a `.gitignore` containing something like: 

```ini
.eslintcache
.stylelintcache
.netlify
.hugo_build.lock
yarn-error.log
node_modules
public
resources
```

And, make sure to run a local build and push the generated `hugo_stats.json` to your repository.

That should do it!

**Reply by torvenkat** [reply]
Thanks.  It worked perfectly.  Much appreciate your kind help.
